py/legacyanalysis/sersic-agn.py

```diff
@@ -0,0 +1,269 @@
+        #cmd = 'ssh cori "cd legacypipe2/py && python legacypipe/runbrick.py --radec %.4f %.4f --width 100 --height 100 --survey-dir fakedr9 --outdir %s --stage image_coadds --skip-calibs && python legacyanalysis/create_testcase.py --survey-dir fakedr9 --outlier-dir %s %s/coadd/*/*/*-ccds.fits %s %s"' % (ra, dec, outdir, outdir, outdir, datadir, brickname)
+
+        outbrick = ('custom-%06i%s%05i' %
+                    (int(1000*ra), 'm' if dec < 0 else 'p',
+                     int(1000*np.abs(dec))))
+
+        cmd = 'ssh cori "cd legacypipe2/py && python legacyanalysis/create_testcase.py --survey-dir fakedr9 --outlier-dir %s --outlier-brick %s %s/coadd/*/*/*-ccds.fits %s %s"' % (outdir, outbrick, outdir, datadir, brickname)
+        #os.system(cmd)
+    
+        cmd = 'rsync -arv cori:legacypipe2/py/%s .' % datadir
+        #os.system(cmd)
+        #continue
+
+        survey = LegacySurveyData(datadir)
+    
+        b = Duck()
+        b.ra = ra
+        b.dec = dec
+    
+        W,H = 80,80
+        wcs = wcs_for_brick(b, W=W, H=H)
+        targetrd = np.array([wcs.pixelxy2radec(x,y) for x,y in
+                             [(1,1),(W,1),(W,H),(1,H),(1,1)]])
+        ccds = survey.ccds_touching_wcs(wcs)
+        print(len(ccds), 'CCDs')
+    
+        ims = [survey.get_image_object(ccd) for ccd in ccds]
+        keepims = []
+        for im in ims:
+            h,w = im.shape
+            if h >= H and w >= W:
+                keepims.append(im)
+        ims = keepims
+        gims = [im for im in ims if im.band == 'g']
+        rims = [im for im in ims if im.band == 'r']
+        zims = [im for im in ims if im.band == 'z']
+        nk = min([len(gims), len(rims), len(zims), 5])
+        ims = gims[:nk] + rims[:nk] + zims[:nk]
+        print('Keeping', len(ims), 'images')
+        
+        tims = [im.get_tractor_image(pixPsf=True, hybridPsf=True, normalizePsf=True, splinesky=True, radecpoly=targetrd) for im in ims]
+    
+        bands = 'grz'
+    
+        devsrc = DevGalaxy(RaDecPos(ra,dec), NanoMaggies(**dict([(b,10.) for b in bands])), EllipseESoft(0., 0., 0.))
+        tr = Tractor(tims, [devsrc])
+        tr.freezeParam('images')
+        tr.optimize_loop()
+        print('Fit DeV source:', devsrc)
+        devmods = list(tr.getModelImages())
+        showboth(tims, devmods, wcs, bands);
+        s = flux_string(devsrc.brightness)
+        plt.suptitle('DeV model: ' + s + '\ndchisq 0.')
+        plt.savefig('src%02i-dev.png' % isrc)
+        devchi = 2. * tr.getLogLikelihood()
+        
+        dasrc = DevAgnGalaxy(devsrc.pos.copy(), devsrc.brightness.copy(), devsrc.shape.copy(), NanoMaggies(**dict([(b,1.) for b in bands])))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['DevAgnGalaxy' may be undefined, or defined from star imports: legacypipe.survey, tractor, tractor.devagn, tractor.galaxy](https://app.codacy.com/app/dstndstn/legacypipe/pullRequest?prid=4730106)